src/open_r1/rewards.py

```diff
@@ -384,8 +386,15 @@ def extract_code(completion: str, language: str = "python") -> str:
     return extracted_answer
 
 
-def binary_code_reward(completions, num_parallel: int = 2, e2b_router_url=None, **kwargs) -> list[float]:
-    rewards = code_reward(completions, num_parallel=num_parallel, e2b_router_url=e2b_router_url, **kwargs)
+def binary_code_reward(completions, num_parallel: int = 2, e2b_router_url=None, provider_type: str = None, enforce_same_language: bool = False, **kwargs) -> list[float]:
```

Comment:
I realise this is a breaking change, but maybe it's better to do this than have a router URL arg for E2B and then have users remember they need `kwargs` for Morph:
```suggestion
def binary_code_reward(completions, num_parallel: int = 2, provider_type: str = None, enforce_same_language: bool = False, **kwargs) -> list[float]:
```

The alternative would be to expose